util/logger.go

```diff
@@ -27,75 +31,84 @@ var (
 	// (see https://github.com/gruntwork-io/terragrunt/blob/master/cli/args.go#L29)
 	GlobalFallbackLogEntry *logrus.Entry
 
-	disableLogColors bool
-	jsonLogFormat    bool
+	globalDisableColors     bool
+	globalDisableFormatting bool
+	jsonLogFormat           bool
 )
 
 func init() {
-	defaultLogLevel := GetDefaultLogLevel()
-	GlobalFallbackLogEntry = CreateLogEntry("", defaultLogLevel)
+	logLevel := GetDefaultLogLevel()
+	GlobalFallbackLogEntry = CreateLogEntry("", logLevel, nil, globalDisableColors, globalDisableFormatting)
+}
+
+func updateGlobalLogger() {
+	GlobalFallbackLogEntry = CreateLogEntry("", defaultLogLevel, nil, globalDisableColors, globalDisableFormatting)
 }
 
 func DisableLogColors() {
-	disableLogColors = true
+	globalDisableColors = true
+	// Needs to re-create the global logger
+	updateGlobalLogger()
+}
+
+func DisableLogFormatting() {
+	globalDisableFormatting = true
 	// Needs to re-create the global logger
-	GlobalFallbackLogEntry = CreateLogEntry("", defaultLogLevel)
+	updateGlobalLogger()
 }
 
 func JsonFormat() {
 	jsonLogFormat = true
 	// Needs to re-create the global logger
-	GlobalFallbackLogEntry = CreateLogEntry("", defaultLogLevel)
+	updateGlobalLogger()
 }
 
 func DisableJsonFormat() {
 	jsonLogFormat = false
 	// Needs to re-create the global logger
-	GlobalFallbackLogEntry = CreateLogEntry("", defaultLogLevel)
+	updateGlobalLogger()
 }
 
 // CreateLogger creates a logger. If debug is set, we use ErrorLevel to enable verbose output, otherwise - only errors are shown
-func CreateLogger(lvl logrus.Level) *logrus.Logger {
+func CreateLogger(lvl logrus.Level, prefixStyle *formatter.PrefixStyle, disableColors, disableFormatting bool) *logrus.Logger {
```

Comment:
I think this would work better with variadic options:
https://medium.com/soon-london/variadic-configuration-functions-in-go-8cef1c97ce99